### PR TITLE
Specify pyjnius in `requirements.in`

### PR DESCRIPTION
**Pull request by sean-rose**
pynjius originally had to be split out into a separate `java-requirements.in` file because it ran `which javac` when imported, which failed when dependabot tried to import pyjnius when checking for updates (#1854).

Now apparently pynjius no longer does `which javac` so it can be put back in `requirements.in`, simplifying things.

In particular, this will allow the `bqetl dependency record` command in Netlify CI builds to succeed without having to add an extra step to install packages from `java-requirements.txt`.

---

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -952,9 +984,10 @@ six==1.15.0 \
     #   oauth2client
     #   ppft
     #   protobuf
+    #   pyjnius
     #   python-dateutil
     #   virtualenv
-smart-open==6.0.0 \
+smart_open==6.0.0 \
```
Comment (sean-rose) [inline requirements.txt:990]: This is weird.  Running pip-compile locally using Python 3.8 changed this to `smart_open`, which seems correct to me, but the [CircleCI verify-requirments check](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/14333/workflows/bef41822-d4ab-4bb5-880b-ba071acb6567/jobs/116973) is insisting this should be `smart-open`.

File: setup.py [hunk 2]
```diff
@@ -56,7 +56,6 @@ def get_version():
         "stripe",
         "authlib",
     ],
-    extras_require={"java": ["pyjnius"]},
```
Comment (sean-rose) [inline setup.py:59]: Not entirely sure about removing this, but since it was added in #1990 I'm defaulting to taking it out.

**Comment by sean-rose** [comment]
[generate-diff](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/14333/workflows/bef41822-d4ab-4bb5-880b-ba071acb6567/jobs/116986) is failing because it runs commands against the `main` branch, which doesn't yet have pyjnius in `requirements.txt`, so it doesn't get installed and the `jnius` module can't be found.

File: requirements.txt (hunk 1 above)
Comment (relud) [reply requirements.txt:990]: might be related to osx vs linux, but we should stick with `smart-open` if that's what CI gives us.

**Review by relud — APPROVED** [review]
r+wc

File: requirements.txt (hunk 1 above)
Comment (sean-rose) [reply requirements.txt:990]: Never mind, looks like I had pip-tools 6.4.0 installed.  Upgrading to pip-tools 6.6.0 and running pip-compile locally changes this to `smart-open` like the CI.

**Comment by sean-rose** [comment]
[integration test](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/14338/workflows/8ed6c362-2449-4dc4-aee1-0b50a3de36fe/jobs/117024) failed with the random, always fun `File /tmp/gcp.json is not a valid json file` error.  Ignoring.

**Comment by sean-rose** [comment]
I've verified that the Netlify CI build can now run `bqetl dependency record` successfully without getting `ModuleNotFoundError: No module named 'jnius'`.

**Pull request merged**